### PR TITLE
Create FST-1004-versioning-plan.md

### PR DESCRIPTION
**Pull request by cartermp**
Initial draft

### CODE REVIEW

File: tooling/FST-1004-versioning-plan.md [hunk 1]
```diff
@@ -0,0 +1,88 @@
...
+* The F# major language version number will change more and more slowly over time.  Proposed and approved new features still fit snugly in the minor version number.
+* The F# minor language version is expected to change, but not as often as tooling and SDK versions.
+* Decoupling the FSharp.Compiler.Service version from the language version has proven to be a good thing, as it has allowed the adoption of SemVer and rapid iteration.  A similar model can be applied for the F# compiler and tooling.
+* Coupling the compiler version with the language version is not always done in other languages (e.g., C#), and such decoupling hasn't been a problem.
+
+## Table of versions
+
+| Visual Studio version | 15.5 (current) | 15.6 | 15.7 | vNext |
+|------------|----------:|-----:|------:|------:|
+| **F# language version** | 4.1 | 4.1 | 5.4 | 4.x |
```
Comment (mexx) [inline tooling/FST-1004-versioning-plan.md]: I like the fact of having new major version of F# so soon, but I think you meant `4.5`, right?

Comment (cartermp) [reply tooling/FST-1004-versioning-plan.md]: Yes, mistype. Thanks!

File: tooling/FST-1004-versioning-plan.md [hunk 2]
```diff
@@ -0,0 +1,88 @@
...
+* The F# minor language version is expected to change, but not as often as tooling and SDK versions.
+* Decoupling the FSharp.Compiler.Service version from the language version has proven to be a good thing, as it has allowed the adoption of SemVer and rapid iteration.  A similar model can be applied for the F# compiler and tooling.
+* Coupling the compiler version with the language version is not always done in other languages (e.g., C#), and such decoupling hasn't been a problem.
+
+## Table of versions
+
+| Visual Studio version | 15.5 (current) | 15.6 | 15.7 | vNext |
+|------------|----------:|-----:|------:|------:|
+| **F# language version** | 4.1 | 4.1 | 4.5 | 4.x |
+| **FSharp.Core version** | 4.4.1.0 | 4.4.3.x | 4.5.0.x | 4.4.x.y |
```
Comment (PatrickMcDonald) [inline tooling/FST-1004-versioning-plan.md]: Should the FSharp.Core version for vNext be 4.x.0.y (or maybe 4.x.y.0)

Comment (cartermp) [reply tooling/FST-1004-versioning-plan.md]: Yep, thanks!

**Pull request merged**